src/Hit/HitAbstract.php

```diff
@@ -296,37 +312,107 @@ public function setSessionNumber($sessionNumber)
         return $this;
     }
 
+    /**
+     * @return string
+     */
+    public function getKey()
+    {
+        return $this->key;
+    }
 
+    /**
+     * @param string $key
+     * @return HitAbstract
+     */
+    public function setKey($key)
+    {
+        $this->key = $key;
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getCreatedAt()
+    {
+        return $this->createdAt;
+    }
 
+    /**
+     * @param string $createdAt
+     * @return HitAbstract
+     */
+    public function setCreatedAt($createdAt)
+    {
+        $this->createdAt = $createdAt;
+        return $this;
+    }
 
     /**
      * Return an associative array of the class with Api parameters as keys
      *
      * @return array
      */
-    public function toArray()
+    public function toApiKeys()
     {
         $data = [
-            FlagshipConstant::VISITOR_ID_API_ITEM => $this->getVisitorId(),
+            FlagshipConstant::VISITOR_ID_API_ITEM => $this->visitorId ?: $this->anonymousId,
             FlagshipConstant::DS_API_ITEM => $this->getDs(),
             FlagshipConstant::CUSTOMER_ENV_ID_API_ITEM => $this->getConfig()->getEnvId(),
             FlagshipConstant::T_API_ITEM => $this->getType(),
             FlagshipConstant::USER_IP_API_ITEM => $this->getUserIP(),
             FlagshipConstant::SCREEN_RESOLUTION_API_ITEM => $this->getScreenResolution(),
             FlagshipConstant::USER_LANGUAGE => $this->getLocale(),
-            FlagshipConstant::SESSION_NUMBER => $this->getSessionNumber()
+            FlagshipConstant::SESSION_NUMBER => $this->getSessionNumber(),
+            FlagshipConstant::CUSTOMER_UID => null
         ];
 
         if ($this->visitorId && $this->anonymousId) {
             $data[FlagshipConstant::VISITOR_ID_API_ITEM] = $this->anonymousId;
             $data[FlagshipConstant::CUSTOMER_UID] = $this->visitorId;
-        } else {
-            $data[FlagshipConstant::VISITOR_ID_API_ITEM] = $this->visitorId ?: $this->anonymousId;
-            $data[FlagshipConstant::CUSTOMER_UID] = null;
         }
         return $data;
     }
 
+    /**
+     * @param $class
+     * @param $data
+     * @return object
+     * @throws ReflectionException
+     */
+    public static function hydrate($class, $data)
```

Comment:
I think you can do this. :

Reply:
```
public function hydrate(array $data) {}

// and later

$object = new Object();
$object->hydrate($data);

or directly in the constructor : 
$object = new Object($data);
````

The class attribute can be removed I guess.
https://stackoverflow.com/questions/46830734/hydrate-and-constructors-in-php